app/src/features/rules/screens/sharedListViewer/hooks/useFetchSharedListData.ts

```diff
@@ -0,0 +1,53 @@
+import { useEffect, useState } from "react";
+import { fetchSharedListData } from "../../sharedLists/utils";
+import { generateGroupwiseRulesMap } from "../utils";
+import Logger from "../../../../../../../common/logger";
+import { Group, Rule } from "types";
+
+interface Props {
+  sharedListId: string;
+}
+
+export const useFetchSharedListData = ({ sharedListId }: Props) => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [sharedListGroupsMap, setSharedListGroupsMap] = useState<Record<string, Group>>({});
+  const [sharedListGroupwiseRulesMap, setSharedListGroupwiseRulesMap] = useState<Record<string, Rule[]>>({});
```

Comment:
No need for this two states, directly generate the objects 
```suggestion
 const sharedListGroupsMap: Record<string, Group>  = useMemo(() => generateGroupwiseRulesMap(sharedListRules), [sharedListRules])
  const sharedListGroupwiseRulesMap: Record<string, Rule[]> = useMemo(() => sharedListGroups.reduce((map: Record<string, Group>, group: Group) => {
  map[group.id] = group;
  return map
  }, {}),[sharedListGroups])
```